### PR TITLE
Update local cli executor to use same filename strategy as docker

### LINKED ISSUE #1978: [Bug]: LocalCommandLineCodeExecutor doesn't save code files
### Describe the bug

When using a LocalCommandLineCodeExecutor and specifying to the agent that the code should be saved to disk and the code contains the pre-requisite "# filename <filename.py>" (for Python inside first line of code block), the file is not saved.

When comparing to the DockerCommandLineCodeExecutor, I see that there is code that parses the generated code block and extracts the file name specified.  See line 176 inside execute_code_blocks()

Is there a reason this feature is missing from the LocalCommandLineCodeExecutor? 

### Steps to reproduce

Write a Python script that uses code generation agents and ask for the code to be saved.  Use the LocalCommandLineCodeExecutor to execute the code using the default system message (which includes the instruction to name the file as first line comment at top of code block)

Run the code and you will notice the comment with filename is generated, but the file isn't saved.  Only the filename with the uuid is saved.

### Model Used

mixtral 8x7b using Ollama.  I don't think the model makes a difference though

### Expected Behavior

Filename should be parsed if it is specified in the first line comment of the generated code block and file should be saved to the working directory specified to the LocalCommandLineCodeExecutor

### Screenshots and logs

_No response_

### Additional Information

AugoGen Version: 0.2.18
Operating System: macOS Sonoma 14.4
Python Version: 3.11.7

### PR DESCRIPTION
**Pull request by jackgerrits**
## Why are these changes needed?

- Additionally, this ensures that both stderr and stdout are returned from the executor

## Related issue number

Closes #1978

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/coding/local_commandline_code_executor.py [hunk 1]
```diff
@@ -150,41 +145,53 @@ def execute_code_blocks(self, code_blocks: List[CodeBlock]) -> CommandLineCodeRe
...
+            logs_all += result.stderr
+            logs_all += result.stdout
+            exitcode = result.returncode
+
             if exitcode != 0:
                 break
 
-        code_filename = str(self._work_dir / filename) if filename is not None else None
-        return CommandLineCodeResult(exit_code=exitcode, output=logs_all, code_file=code_filename)
+        return CommandLineCodeResult(exit_code=exitcode, output=logs_all, code_file=str(file_names[0]))
```
Comment (ekzhu) [inline autogen/coding/local_commandline_code_executor.py:194]: Is it possible the list of filenames is empty?

File: autogen/coding/local_commandline_code_executor.py [hunk 2]
```diff
@@ -150,41 +145,53 @@ def execute_code_blocks(self, code_blocks: List[CodeBlock]) -> CommandLineCodeRe
...
+                filename = f"tmp_code_{code_hash}.{'py' if lang.startswith('python') else lang}"
+
+            written_file = (self._work_dir / filename).resolve()
+            written_file.open("w", encoding="utf-8").write(code)
+            file_names.append(written_file)
+
+            program = sys.executable if lang.startswith("python") else _cmd(lang)
+            cmd = [program, str(written_file.absolute())]
+
+            try:
```
Comment (ekzhu) [inline autogen/coding/local_commandline_code_executor.py:177]: Is there a reason not to use the ThreadPoolExecutor as in the `code_utils`?

Comment (jackgerrits) [reply autogen/coding/local_commandline_code_executor.py:182]: Why would we use a `ThreadPoolExecutor`? It wasn't necessary

File: autogen/coding/local_commandline_code_executor.py (hunk 1 above)
Comment (jackgerrits) [reply autogen/coding/local_commandline_code_executor.py:199]: true, let me check for that

**Review by gagb — APPROVED** [review]

**Comment by jackgerrits** [comment]
@sonichi can you retrigger tests?

**Comment by sonichi** [comment]
> @sonichi can you retrigger tests?

Please make the PR from upstream.

**Review by ekzhu — APPROVED** [review]

**Pull request closed without merging**